src/porsas/next.clj

```diff
@@ -23,9 +23,17 @@
            rs/RowBuilder
            (->row [_] (->row rs))
            (with-column [_ row _] row)
+           (with-column-value [_ row _ _] row)
            (column-count [_] 0)
            (row! [_ row] row)
            rs/ResultSetBuilder
            (->rs [_] (transient []))
            (with-row [_ rs row] (conj! rs row))
-           (rs! [_ rs] (persistent! rs))))))))
+           (rs! [_ rs] (persistent! rs))
+           clojure.lang.ILookup ; only supports :cols and :rsmeta
+           (valAt [this k] (get this k nil))
+           (valAt [this k not-found]
+             (case k
+               :cols   (map last (pj/col-map rs key))
```

Comment:
Could this be an empty vector as `column-count` is always `0`?

Reply:
The only thing `column-count` is used for internally is iterating over the (number of columns) in the internal `row-builder` function, which I don't believe is called in the case of the `porsas` code? (or maybe it is done to short-circuit that method so much of the internals of `next.jdbc` can be skipped over?)

The actual column names do exist and `(map last (pj/col-map rs key))` looks like the way `porsas` fetches those when building the row.

Reply:
I believe this `row-builder` exists to do some optimizations on how `next.jdbc` builds up the results (at least that's my understanding based on the test [here](https://github.com/metosin/porsas/blob/0a803db1e689a120babe7b11138e4b866c715050/test/porsas/core_test.clj#L93). So if the `column-count` is constantly `0` then the `cols` here won't be used so no need to calculate them, although it is lazy so won't be calculated unnecessarily.